src/rime_api.cc

```diff
@@ -389,20 +389,26 @@ RIME_API Bool RimeFreeStatus(RimeStatus* status) {
 
 // Accessing candidate list
 
+RIME_API Bool RimeCandidateListFromIndex(RimeSessionId session_id,
+                                         RimeCandidateListIterator* iterator,
+                                         int index) {
+    if (!iterator)
+        return False;
+    an<Session> session(Service::instance().GetSession(session_id));
+    if (!session)
+        return False;
+    Context *ctx = session->context();
+    if (!ctx || !ctx->HasMenu())
+        return False;
+    memset(iterator, 0, sizeof(RimeCandidateListIterator));
+    iterator->ptr = ctx->composition().back().menu.get();
+    iterator->index = index - 1;
+    return True;
```

Comment:
It would be better to conform to indentation style elsewhere (2 spaces instead of 4).